Sources/SwiftParser/generated/LayoutNodes+Parsable.swift

```diff
@@ -345,4 +363,8 @@ fileprivate extension Parser {
     }
     return node
   }
+  
+  mutating func parseVersionTupleWithFourComponents() -> RawVersionTupleSyntax {
+    return self.parseVersionTuple(maxComponentCount: 4)
```

Comment:
I’m not a huge fan of this. 4 seems as arbitrary a value as any other, so we shouldn’t default to it. And actually, since we are moving more and more of the version handling logic out of the parser and into higher-level libraries, I’m thinking that we should remove the `maxComponentCount` parameter altogether. If, e.g. the IfConfig library can only handle 4 components, it should be responsible for diagnosing this, instead of the parser.

So, my suggestion is to remove the `maxComponentCount` parameter, even if it means that some of the current diagnostics will regress. We have the expectation that these diagnostics will be emitted by the libraries that interpret the version tuple.

Reply:
That makes sense to me👍.